sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -2487,21 +2487,29 @@ internal virtual void SetNameFieldsIfNull()
 
                         do
                         {
-                            jsonResponse =
-                                await DataLakeRestClient.Path.SetAccessControlRecursiveAsync(
-                                    clientDiagnostics: ClientDiagnostics,
-                                    pipeline: Pipeline,
-                                    resourceUri: DfsUri,
-                                    mode: mode,
-                                    maxRecords: options.BatchSize,
-                                    version: Version.ToVersionString(),
-                                    acl: accessControlList,
-                                    async: async,
-                                    continuation: continuationToken,
-                                    forceFlag: options.ContinueOnFailure ? true : (bool?)default,
-                                    cancellationToken: cancellationToken)
-                                .ConfigureAwait(false);
-
+                            try
+                            {
+                                jsonResponse =
+                                    await DataLakeRestClient.Path.SetAccessControlRecursiveAsync(
+                                        clientDiagnostics: ClientDiagnostics,
+                                        pipeline: Pipeline,
+                                        resourceUri: DfsUri,
+                                        mode: mode,
+                                        maxRecords: options.BatchSize,
+                                        version: Version.ToVersionString(),
+                                        acl: accessControlList,
+                                        async: async,
+                                        continuation: continuationToken,
+                                        forceFlag: options.ContinueOnFailure ? true : (bool?)default,
+                                        cancellationToken: cancellationToken)
+                                    .ConfigureAwait(false);
+                            }
+                            catch (RequestFailedException storageRequestFailedException)
+                            when (!string.IsNullOrEmpty(continuationToken) &&
+                                storageRequestFailedException.Status >= Constants.HttpStatusCode.ServerError)
+                            {
+                                throw new OperationInterruptedException(storageRequestFailedException, continuationToken);
```

Comment:
should we document this in the API docs somehow and/or provide sample ? so that customer is aware they should be catching/checking subtype?

Reply:
Let's look to add the sample later. I made an issue for it. https://github.com/Azure/azure-sdk-for-net/issues/15409